### PR TITLE
Read `window.rAF` later to allow fake timers to work correctly

### PR DESCRIPTION
**Pull request by ensconced**
This fixes issue https://github.com/reduxjs/redux-toolkit/issues/4693 by reading window.requestAnimationFrame within the autoBatchEnhancer function instead of in the top-level module scope. This ensures that we use the fake version of requestAnimationFrame if jest is using fake timers.

### CODE REVIEW

**Comment by ensconced** [comment]
I'm just trying to figure out how I can write a test for this

File: packages/toolkit/src/autoBatchEnhancer.ts [hunk 1]
```diff
@@ -62,6 +55,13 @@ export const autoBatchEnhancer =
 
     const listeners = new Set<() => void>()
 
+    // requestAnimationFrame won't exist in SSR environments.
+    // Fall back to a vague approximation just to keep from erroring.
+    const rAF =
+      typeof window !== 'undefined' && window.requestAnimationFrame
+        ? window.requestAnimationFrame
+        : createQueueWithTimer(10)
+
```
Comment (phryneas) [inline packages/toolkit/src/autoBatchEnhancer.ts:64]: At this point, this could probably be inlined in line 69.

Comment (ensconced) [reply packages/toolkit/src/autoBatchEnhancer.ts]: Makes sense, I have pushed this change.

File: packages/toolkit/src/tests/autoBatchEnhancer.test.ts [hunk 2]
```diff
@@ -125,3 +125,84 @@ describe.each(cases)('autoBatchEnhancer: %j', (autoBatchOptions) => {
     expect(subscriptionNotifications).toBe(3)
   })
 })
+
+describe.each(cases)(
```
Comment (ensconced) [inline packages/toolkit/src/tests/autoBatchEnhancer.test.ts:129]: Two of these tests were failing before the fix was applied

File: packages/toolkit/src/autoBatchEnhancer.ts [hunk 3]
```diff
@@ -66,7 +59,8 @@ export const autoBatchEnhancer =
       options.type === 'tick'
         ? queueMicrotask
         : options.type === 'raf'
-          ? rAF
+          ? // requestAnimationFrame won't exist in SSR environments. Fall back to a vague approximation just to keep from erroring.
+            window?.requestAnimationFrame ?? createQueueWithTimer(10)
```
Comment (phryneas) [inline packages/toolkit/src/autoBatchEnhancer.ts:63]: `window?` doesn't cover all cases that `typeof window !== 'undefined'` does - optional chaining still can cause a reference error if a variable with that name never was defined:
grafik

Comment (ensconced) [reply packages/toolkit/src/autoBatchEnhancer.ts]: ah, of course. I have just pushed a fix to revert it back to a ternary like before

**Comment by ensconced** [comment]
Let me know if there's anything else I can do to help this get merged 🙂

**Comment by phryneas** [comment]
Mostly, "make the day have 48 hours" :)

Frankly, we have a ton to do (here and in our day-to-day lifes), so PRs can sit around for quite a while until one of us finds the time to thoroughly go over a few of them.

At first glance, this looks good, but please don't expect this to be merged/released soon. You can use the CodeSandbox release from the CSB comment in the meantime!

**Pull request merged**